docs/advanced_usages/custom-tests.md

```diff
@@ -0,0 +1,218 @@
+# Create your own custom tests
+
+ANTA is not only a CLI with a collection of built-in tests, it is also a framework you can extend by building your own tests library.
+
+For that, you need to create your own Python package as descirbed in this [hitchhiker's guide](https://the-hitchhikers-guide-to-packaging.readthedocs.io/en/latest/) to package Python code. We assume it is well known and we won't focus on this aspect.
+
+## Generic approach
+
+To make it easier for everyone, it is recommended to apply always same structure to all code:
+
+```python
+
+# Optional for easy coding
+from typing import Optional
+import logging
+
+# Import ANTA components
+from anta.inventory.models import InventoryDevice
+from anta.result_manager.models import TestResult
+from anta.tests import anta_test
+
+# Instantiate logger
+logger = logging.getLogger(__name__)
+
+# define a function with at least one decorator to mark as a test
+@anta.test
+async def verify_dynamic_vlan(device: InventoryDevice,
+    result: TestResult, minimum: Optional[int] = None) -> TestResult:
+
+    # Create docstring
+    # [...]
+
+    # Create your own logic
+
+    # Check if test option is correct
+    if not minimum:
+        result.is_skipped("verify_dynamic_vlan was run without minimum value set")
+        return result
+
+    # Send command to device
+    response = await device.session.cli(command="show vlan", ofmt="json")
+
+    # Optionaly log response for futur debug
+    logger.debug(f"query result is: {response}")
+
+    # Do your test: In this example we count number of vlans with field dynamic set to true
+    num_dyn_vlan = len([ vlan for vlan,data in response['vlans'].items() if data['dynamic'] is True])
+    if num_dyn_vlan >= minimum:
+        result.is_success()
+    else:
+        result.is_failure(f"Device has {num_dyn_vlan} configured, we expect at least {minimum}")
+
+    return result
+```
+
+## Python imports
+
+### Mandatory imports
+
+You should import at least following elements:
+
+- [`InventoryDevice`](../api/inventory.models.md): Where the eAPI session lives. It is used to send commands over HTTP/HTTPS define in your test.
+- [`TestResult`](../api/result_manager_models.md): Structure used to facilitate test result. It provides helpers like `is_success`, `is_failure`, `is_skipped` and also runs execption managment for you.
```

Comment:
Though is the test result really running exception management? the decorator does that no?
```suggestion
- [`TestResult`](../api/result_manager_models.md): Structure used to facilitate test result. It provides helpers like `is_success`, `is_failure`, `is_skipped` and also runs exception management for you.
```

Reply:
My understanding is we capture exception in this decorator

https://github.com/arista-netdevops-community/anta/blob/7b1640c56cc474595f3b6366ac329fc2f6abb2ac/anta/tests/__init__.py#L44

Reply:
yes abd si TestResut does not run exception management right?

Reply:
Oh yes, my bad it is in the wrong place. Fixing it shortly